### ISSUE TITLE
error in loading vlc in Mac

### ISSUE THREAD

**Issue by Abu-Abdullah** [caprica/vlcj#367]
Hi,

for some reason, i start getting this error in Mac OS 10.10 after formatting the machine:
```
java.lang.UnsatisfiedLinkError: Unable to load library 'vlc': JNA native support (darwin/libvlc.jnilib) not found in resource path  ...............
````

The difference is libvlc.jnilib instead of libvlc.dylib which is in the correct directory.

I'm using 64 bit of java and vlc 2.2 64bit. setting VLC_PLUGIN_PATH at the beginning and NativeLibrary.addSearchPath to the set the lib path.

i tried many things e.g. vlcj 2.5/2.4/2.6, jna 2.5.2/4.1.0, vlc 2.2/2.1.5, java 8_u40/u31 

any help is appreciated

**Comment by caprica** [comment]
On this one I have no idea...

**Comment by Abu-Abdullah** [comment]
i tried to create sample code which is:
```
import java.awt.event.WindowAdapter;
import java.awt.event.WindowEvent;

import javax.swing.JFrame;
import javax.swing.SwingUtilities;

import uk.co.caprica.vlcj.component.EmbeddedMediaPlayerComponent;
import uk.co.caprica.vlcj.discovery.NativeDiscovery;

import uk.co.caprica.vlcj.binding.*;
import uk.co.caprica.vlcj.binding.internal.libvlc_media_t;
import uk.co.caprica.vlcj.runtime.RuntimeUtil;
import uk.co.caprica.vlcj.player.*;
import uk.co.caprica.vlcj.binding.LibVlcConst;
import com.sun.jna.*;

public class Tutorial implements MediaPlayerEventListener
{

    private final JFrame frame;

    private final MediaPlayerFactory mediaPlayerComponent;
	

    public static void main(final String[] args) {
		uk.co.caprica.vlcj.binding.LibC.INSTANCE.setenv("VLC_PLUGIN_PATH", "./vlc/Contents/MacOS/plugins", 1);
		NativeLibrary.addSearchPath(RuntimeUtil.getLibVlcLibraryName(), "./vlc/Contents/MacOS/lib");
		Native.loadLibrary(RuntimeUtil.getLibVlcLibraryName(), LibVlc.class);
        SwingUtilities.invokeLater(new Runnable() {
            @Override
            public void run() {
                new Tutorial(args);
            }
        });
    }

    public Tutorial(String[] args) 
	{
        frame = new JFrame("My First Media Player");
        frame.setBounds(100, 100, 600, 400);
        frame.setDefaultCloseOperation(JFrame.DO_NOTHING_ON_CLOSE);
        frame.addWindowListener(new WindowAdapter() {
            @Override
            public void windowClosing(WindowEvent e) {
                mediaPlayerComponent.release();
                System.exit(0);
            }
        });
		
		String[] VLC_ARGS = {
				"--intf=dummy",          	// no interface
				"--vout=dummy",          	// we don't want video (output)
				"--no-video-title-show",    // nor the filename displayed
				"--no-stats",               // no stats
				"--no-sub-autodetect-file", // we don't want subtitles
				"--no-disable-screensaver", // we don't want interfaces
				"--no-snapshot-preview",    // no blending in dummy vout
				"--no-plugins-cache",		// for accurate seek via internet
				"--lua-intf=dummy"
				//"--quiet",				// Slow start playback !
				//"--quiet-synchro",
				//"--file-caching=0",		// No effect
				//"--disc-caching=0",		// No effect
				//"--network-caching=0",
				//"--codec=ffmpeg",
		};
		
        mediaPlayerComponent = new MediaPlayerFactory(VLC_ARGS);
        //frame.setContentPane(mediaPlayerComponent);
        //frame.setVisible(true);
        mediaPlayerComponent.newHeadlessMediaPlayer().playMedia("test.mp3");
    }
	
	public void timeChanged(MediaPlayer mediaPlayer, long newTime)
    {
        
    }

    //  @Override
    public void scrambledChanged(MediaPlayer mediaPlayer, int newScrambled) {
    }

    //  @Override
    public void elementaryStreamDeleted(MediaPlayer mediaPlayer, int type, int id) {
    }

    //  @Override
    public void elementaryStreamSelected(MediaPlayer mediaPlayer, int type, int id) {
    }

    //  @Override
    public void elementaryStreamAdded(MediaPlayer mediaPlayer, int type, int id) {
    }

    //  @Override
    public void mediaChanged(MediaPlayer mediaPlayer, libvlc_media_t media, String mrl) {
    }

    //  @Override
    public void mediaSubItemTreeAdded(MediaPlayer mediaPlayer, libvlc_media_t item) {
    }

    //  @Override
    public void opening(MediaPlayer mediaPlayer) {
    }

    //  @Override
    public void buffering(MediaPlayer mediaPlayer, float newCache)
	{
		
    }

    //  @Override
	//public void playing(MediaPlayer mediaPlayer){} // Before buffering
    public void playing(MediaPlayer mediaPlayer)
	{
		
    }

    //  @Override
    public void paused(MediaPlayer mediaPlayer)
    {
		
    }

    //  @Override
    public void stopped(MediaPlayer mediaPlayer)
	{
		
    }

    //  @Override
    public void forward(MediaPlayer mediaPlayer) {
    }

    //  @Override
    public void backward(MediaPlayer mediaPlayer) {
    }

    //  @Override
    public void finished(MediaPlayer mediaPlayer)
	{
		
    }

    //  @Override
    public void positionChanged(MediaPlayer mediaPlayer, float newPosition) {
    }

    //  @Override
    public void seekableChanged(MediaPlayer mediaPlayer, int newSeekable) {
    }

    //  @Override
    public void pausableChanged(MediaPlayer mediaPlayer, int newSeekable) {
    }

    //  @Override
    public void titleChanged(MediaPlayer mediaPlayer, int newTitle) {
    }

    //  @Override
    public void snapshotTaken(MediaPlayer mediaPlayer, String filename) {
    }

    //  @Override
    public void lengthChanged(MediaPlayer mediaPlayer, long newLength) {
    }

    //  @Override
    public void videoOutput(MediaPlayer mediaPlayer, int newCount) {
    }

    //  @Override
    public void error(MediaPlayer mediaPlayer)
	{
		
    }

    //  @Override
    public void mediaMetaChanged(MediaPlayer mediaPlayer, int metaType) {
    }

    //  @Override
    public void mediaSubItemAdded(MediaPlayer mediaPlayer, libvlc_media_t subItem) {
    }

    //  @Override
    public void mediaDurationChanged(MediaPlayer mediaPlayer, long newDuration) {
    }

    //  @Override
    public void mediaParsedChanged(MediaPlayer mediaPlayer, int newStatus) {
    }

    //  @Override
    public void mediaFreed(MediaPlayer mediaPlayer) {
    }

    //  @Override
    public void mediaStateChanged(MediaPlayer mediaPlayer, int newState) {
    }

    //  @Override
    public void newMedia(MediaPlayer mediaPlayer) {
    }

    //  @Override
    public void subItemPlayed(MediaPlayer mediaPlayer, int subItemIndex) {
    }

    //  @Override
    public void subItemFinished(MediaPlayer mediaPlayer, int subItemIndex) {
    }

    //  @Override
    public void endOfSubItems(MediaPlayer mediaPlayer) {
    }
}
```

compiling with jna 3.5.2:

```


SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".

SLF4J: Defaulting to no-operation (NOP) logger implementation

SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.

Exception in thread "main" java.lang.UnsatisfiedLinkError: Unable to load library 'vlc': JNA native support (darwin/libvlc.jnilib) not found in resource path (.:vlc/Contents/MacOS/lib:jna-3.5.2.jar:platform-3.5.2.jar:slf4j-api-1.7.10.jar:vlcj.jar)

at com.sun.jna.NativeLibrary.loadLibrary(NativeLibrary.java:220)

at com.sun.jna.NativeLibrary.getInstance(NativeLibrary.java:322)

at com.sun.jna.Library$Handler.<init>(Library.java:142)

at com.sun.jna.Native.loadLibrary(Native.java:387)

at com.sun.jna.Native.loadLibrary(Native.java:366)

at Tutorial.main(Tutorial.java:28)
```


now using the latest jna builds from:
https://github.com/twall/jna/blob/master/dist/jna-platform.jar
https://github.com/twall/jna/blob/master/dist/jna.jar

and got new errors now:

```
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".

SLF4J: Defaulting to no-operation (NOP) logger implementation

SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.

Exception in thread "main" java.lang.UnsatisfiedLinkError: dlopen(/Users/oracle/Desktop/test/vlc/Contents/MacOS/lib/libvlc.dylib, 9): no suitable image found.  Did find:

/Users/oracle/Desktop/test/vlc/Contents/MacOS/lib/libvlc.dylib: file too short

at com.sun.jna.Native.open(Native Method)

at com.sun.jna.Native.open(Native.java:1761)

at com.sun.jna.NativeLibrary.loadLibrary(NativeLibrary.java:260)

at com.sun.jna.NativeLibrary.getInstance(NativeLibrary.java:398)

at com.sun.jna.Library$Handler.<init>(Library.java:147)

at com.sun.jna.Native.loadLibrary(Native.java:414)

at com.sun.jna.Native.loadLibrary(Native.java:393)

at Tutorial.main(Tutorial.java:28)
```

**Comment by caprica** [comment]
In the vlcj test sources, find NativeDiscoveryTest and run it.

I just did that on a fully updated OSX 10.10.2 with latest release version VLC 2.2.0 and it worked just fine.

You need to use latest vlcj 3.7.0 snapshot from github because it has the setenv fix for VLC_PLUGIN_PATH on OSX - vlcj 3.7.0 has not yet been released to Maven Central repository.

**Comment by Abu-Abdullah** [comment]
Ok, it turns out to be a corrupted VLC binary. i installed new VLC and it starts to work out.

sorry for wasting efforts in this case, error messages were confusing.
